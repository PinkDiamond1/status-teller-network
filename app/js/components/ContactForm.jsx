import React, {Component, Fragment} from 'react';
import {FormGroup, Label} from 'reactstrap';
import Form from 'react-validation/build/form';
import Input from 'react-validation/build/input';
import {withNamespaces} from "react-i18next";
import PropTypes from 'prop-types';
import {required} from "../validators";

class ContactForm extends Component {
  render() {
    const {t, username, statusContactCode} = this.props;

    return (
      <Fragment>
        <h2>{t('contactForm.yourName')}</h2>
        <p>{t('contactForm.bestWay')}</p>

        <Form>
          <FormGroup>
            <Label for="nickname">Nickname</Label>
            <Input type="text"
                   name="nickname"
                   id="nickname"
                   value={username}
                   className="form-control"
                   onChange={(e) => this.props.changeUsername(e.target.value)}
                   validations={[required]}/>
          </FormGroup>
          <FormGroup>
            <Label for="contactCode">Status contact code or Status ENS name</Label>
            <Input type="text"
                   name="contactCode"
                   id="contactCode"
                   value={statusContactCode}
                   className="form-control"
                   onChange={(e) => this.props.changestatusContactCode(e.target.value)}
                   validations={[required]}/>
          </FormGroup>
        </Form>
      </Fragment>
    );
  }
}

ContactForm.propTypes = {
  t: PropTypes.func,
  changeUsername: PropTypes.func,
  changestatusContactCode: PropTypes.func,
  username: PropTypes.string,
  statusContactCode: PropTypes.string
};


export default withNamespaces()(ContactForm);
