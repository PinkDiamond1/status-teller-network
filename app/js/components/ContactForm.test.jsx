import React from "react";
import { shallow } from "enzyme";

import ContactForm from "./ContactForm";

describe('SellerContact', () => {
  it('should render correctly', () => {
    const component = shallow(<ContactForm changeNickname={() => undefined} changeContactCode={() => undefined}
                                           nickname={"nickname"} contractCode={"contractCode"} />);

    expect(component).toMatchSnapshot();
  });
});
