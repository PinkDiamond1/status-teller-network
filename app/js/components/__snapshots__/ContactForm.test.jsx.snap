// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SellerContact should render correctly 1`] = `
<Fragment>
  <h2 />
  <p />
  <Form(t)>
    <FormGroup
      tag="div"
    >
      <Label
        for="nickname"
        tag="label"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        Nickname
      </Label>
      <Control(s)
        className="form-control"
        id="nickname"
        name="nickname"
        onChange={[Function]}
        type="text"
        validations={
          Array [
            [Function],
          ]
        }
        value="nickname"
      />
    </FormGroup>
    <FormGroup
      tag="div"
    >
      <Label
        for="contactCode"
        tag="label"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        Status contact code or Status ENS name
      </Label>
      <Control(s)
        className="form-control"
        id="contactCode"
        name="contactCode"
        onChange={[Function]}
        type="text"
        validations={
          Array [
            [Function],
          ]
        }
      />
    </FormGroup>
  </Form(t)>
</Fragment>
`;
